### ISSUE TITLE
Min/Max Handling

### ISSUE THREAD

**Issue by dietzc** [knime-ip/knip#308]
Many of our nodes (including IJMacro, Splitter, Merger, ...) don't handle the newly introduced min/max properly.

**Comment by dietzc** [comment]
first move into right direction: https://github.com/knime-ip/knip/commit/94debf78d47b0f159ea971f355adbd8d4c17906f.

Open question: how should the output look like. which leads to: how to handle min/max in general?

**Comment by hornm** [comment]
My preference: ignoring the minimum during the processing (i.e. translating it to the origin), but 're-translating' the result, if possible

**Comment by dietzc** [comment]
yep. I totally agree. However, do we want to implement this strategy in each and every `Op` or `Node` or do we want to have a more general mechanism for that? And: Do we want to take care about this in KNIP1 or will we leave it as (potentially lost minimum) and come up with a better and consistent strategy in KNIP2?

**Comment by hornm** [comment]
I'd be ok with the ad hoc-solution for KNIP1 (as long as we fix it for every affected node). And, yes, I agree, we'll need a better strategy for KNIP2.

**Comment by dietzc** [comment]
>  (as long as we fix it for every affected node)
basically all nodes are affected ;-)

**Comment by hornm** [comment]
Ok, then maybe the other way around: what was the advantage of introducing the virtual translation to the minimum on the getImgPlus()-call?

**Comment by dietzc** [comment]
because we are consistent with the definitions of `Interval`s in `ImgLib2` (e.g. if we crop segments). Before that we were working with `offset` and the `Interval`s started always origin zero. However, everywhere else in `ImgLib2`, `ImageJ-Ops` etc. we use `Interval` `min` and `max`.

So here is what I suggest: As we didn't care before in `KNIP1` and we started caring now, we can simply offer both (see most recent commit, `getZeroMinImgPlus()`). However, the interval min/max is lost if a node doesn't properly handle it. In `KNIP2` we will make it as suggested: translate the `Interval`back if `min!=0`. However, this should happen on `ImageJ-Ops` level.

**Comment by imagejan** [comment]
@dietzc @hornm Just a short question, because I'm currently using offsets in metadata and GroupBy to create montages of many images in a few high-content screening workflows: will the current method with offsets continue to work? Or should I adapt my workflows to be more future-proof?

**Comment by dietzc** [comment]
@imagejan everything will work as before, if it doesn't we didn't do a good job and have to fix it. We have ~700 testflows running each and every night that things stay the same with KNIP 1.x.x.

Things are different with KNIP2 (which we hope to release in the 4th quarter 2016). This will be an entirely new concept (entirely based on ImageJ-Ops, SciJava, etc). However, we will keep KNIP1 such that user can use both in parallel and potentially even use both frameworks in combination. 

In short: If something breaks in KNIP1, let us know. That's a bug :-)

**Comment by dietzc** [comment]
I will commit a fix to the min/max problem tonight, following what @hornm and me discussed above.

**Comment by dietzc** [comment]
see: https://github.com/knime-ip/knip/commit/880f27d2da054bd335c406130f19e27c3215b2e7

for KNIP2 we also have to discuss (in ops) how to handle `Binary`, `Tenary` etc ops if the incoming intervals have different `min`. Example: the `Image Calculator` should then operate on global coordinates, rather than moving everything to `zeroMin` etc.
